chain/ethereum/src/data_source.rs

```diff
@@ -372,6 +383,17 @@ impl DataSource {
                 .iter()
                 .find(move |handler| handler.filter == Some(BlockHandlerFilter::Call))
                 .cloned(),
+            EthereumBlockTriggerType::Polling(_address, trigger_interval) => self
+                .mapping
+                .block_handlers
+                .iter()
+                .find(move |handler| match handler.filter {
+                    Some(BlockHandlerFilter::Polling { every: interval }) => {
+                        interval == *trigger_interval
+                    }
+                    _ => false,
+                })
+                .cloned(),
```

Comment:
Seems to me the correct matching logic would be to check that `(trigger.block - self.start_block) % handler.filter.every == 0`.